datafusion/functions/src/string/ascii.rs

```diff
@@ -103,19 +103,22 @@ impl ScalarUDFImpl for AsciiFunc {
 
 fn calculate_ascii<'a, V>(array: V) -> Result<ArrayRef, ArrowError>
 where
-    V: ArrayAccessor<Item = &'a str>,
+    V: StringArrayType<'a, Item = &'a str>,
 {
-    let iter = ArrayIter::new(array);
-    let result = iter
-        .map(|string| {
-            string.map(|s| {
-                let mut chars = s.chars();
-                chars.next().map_or(0, |v| v as i32)
-            })
+    let values: Vec<_> = (0..array.len())
+        .map(|i| {
+            if array.is_null(i) {
+                0
```

Comment:
Would it be legal to omit this `if` (and the jump it implies)?

Reply:
It would be legal. I am not sure it would be faster, though we could easily check with a benchmark once this PR is merged

We could at least add a check on the outside of the loop

```rust
// If the array has no nulls, skip checking each value
if array.null_count() == 0 {
  let values: Vec<_> = (0..array.len())....
}

Reply:
I had to look it up

> Offsets must be monotonically increasing, that is offsets[j+1] >= offsets[j] for 0 <= j < length, even for null slots. This property ensures the location for all values is valid and well defined.

Reply:
I realize now the spec doesn't say anything about the offset pointing to valid utf8 or not. 
But `value(i)` being not unsafe, I guess `arrow-rs` should provide that guarantee.

Reply:
That is my understanding as well -- it should be safe (in the rust sense) to look up an entry that is set to null